agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -113,6 +112,25 @@ PersistedFile loadTelemetriesFromDisk() {
         tempFile, new String(ikeyBytes, UTF_8), ByteBuffer.wrap(telemetryBytes));
   }
 
+  // reads bytes from a FileInputStream and allocates those into the buffer array byteArray.
+  private static void readFully(
+      FileInputStream fileInputStream, byte[] byteArray, int offset, int length)
+      throws IOException {
+    if (length < 0) {
+      throw new IndexOutOfBoundsException();
+    }
+
+    int i = 0;
```

Comment:
I think would help readability
```suggestion
    int totalRead = 0;
```